### PR TITLE
Fix #11

### LINKED ISSUE #11: (adjoint) Jacobi fields for non mutating manifolds
For manifolds that do not do mutating operations like `Circle` (adjoint) jacobi fields currently fail.

Example

```julia
using Manifolds, ManifoldsDiff
M = Circle()
p = π/2
q = 0.
X = 1.
Y = ManifoldDiff.adjoint_Jacobi_field(M, p, q, 0.5, X, ManifoldDiff.βdifferential_shortest_geodesic_startpoint)
```

### PR DESCRIPTION
**Pull request by mateuszbaran**
This fixes #11 .

There is some problem with deployment of docs that I can't figure out :confused: .

### CODE REVIEW

**Comment by mateuszbaran** [comment]
I think we need `AbstractMetric` and `EuclideanMetric` in ManifoldsBase.jl to be able to define `riemannian_gradient` without depending on Manifolds.jl. What do you think @kellertuer  ?

**Comment by kellertuer** [comment]
Sounds good to me.

**Comment by kellertuer** [comment]
We have the complicated setup here with the label to activate docs - it seems.

**Comment by kellertuer** [comment]
Ah GH-Pages might be another reason, but you could also simplify the documenter action to really always build the docs :)

**Comment by mateuszbaran** [comment]
I think what we might have been missing is DOCUMENTER_KEY set up.

File: src/manifolds.jl [hunk 1]
```diff
@@ -31,6 +31,12 @@ function adjoint_Jacobi_field!(M::ProductManifold, Y, p, q, t, X, β::Tβ) where
     )
     return Y
 end
+function adjoint_Jacobi_field(::Circle{ℝ}, p, q, t, X, β::Tβ) where {Tβ}
```
Comment (kellertuer) [inline src/manifolds.jl:34]: What about Jacobi fields?

Comment (mateuszbaran) [reply src/manifolds.jl:34]: Jacobi fields were already defined for `Circle` and `Euclidean`.

Comment (mateuszbaran) [reply src/manifolds.jl:34]: See here: https://github.com/JuliaManifolds/ManifoldDiff.jl/blob/324dd64b0144524699379ec8ecce15e60e236da1/src/manifolds.jl#L100-L105 .

Comment (kellertuer) [reply src/manifolds.jl:34]: Oh, I must have missed those, sorry :)

**Review by kellertuer — APPROVED** [review]

File: src/manifolds.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/manifolds.jl:34]: No problem :)

**Pull request merged**